File: frontend/golf-directory/src/lib/video-of-the-day-query.js
/**
 * Video of the Day Query
 * Shared between frontend and backend to ensure consistent selection logic
 */

const fs = require('fs');
const path = require('path');

// Read the SQL query from shared file (single source of truth)
const getVideoOfTheDayQuery = () => {
  // Use the shared query file that both frontend and backend use
  const queryPath = path.join(process.cwd(), '..', '..', 'shared', 'video-of-the-day-query.sql');
  return fs.readFileSync(queryPath, 'utf8');
};

module.exports = {
  getVideoOfTheDayQuery
};

File: shared/video_of_the_day_query.py
"""
Video of the Day Query
Shared between frontend and backend to ensure consistent selection logic
"""

import os

def get_video_of_the_day_query():
    """Read the shared video of the day SQL query"""
    query_path = os.path.join(os.path.dirname(__file__), 'video-of-the-day-query.sql')
    with open(query_path, 'r') as f:
        return f.read()